src/main/kotlin/io/hsar/wh40k/combatsimulator/logic/TacticalActionStrategy.kt

```diff
@@ -2,9 +2,24 @@ package io.hsar.wh40k.combatsimulator.logic
 
 import io.hsar.wh40k.combatsimulator.model.UnitInstance
 import io.hsar.wh40k.combatsimulator.model.World
+import io.hsar.wh40k.combatsimulator.model.unit.BaseStat
+import io.hsar.wh40k.combatsimulator.model.unit.StatUtils.getBonus
 
 object TacticalActionStrategy {
     fun decideTurnActions(world: World, thisUnit: UnitInstance): List<TurnAction> {
         TODO("Not yet implemented")
     }
+
+    fun canHalfMoveToEnemy(world: World, thisUnit: UnitInstance, enemy: UnitInstance): Boolean {
+        return  world.distanceApart(thisUnit, enemy)  - 1 <= // -1 as only need to be next to them, not on same square
+                thisUnit.unit.stats.baseStats.getValue(BaseStat.AGILITY).getBonus()
+    }
+
+    fun canChargeToEnemy(world: World, thisUnit: UnitInstance, enemy: UnitInstance): Boolean {
```

Comment:
I think these two methods would be nicer as a generic `canMoveToUnit()` method that takes a `MoveAction` (where `MoveAction` is a new interface like `DamageCausingAction` that is implemented by two new `TurnActions` `HalfMove` and `Charge`. That would allow us to add the missing move actions `FullMove` and `Run` simply by extending `canMoveToUnit()`.

Can you make that change?

Reply:
Yep, wilco.

Reply:
My suggestion is to implement it with a `when` clause, which is a super awesome Kotlin feature that makes good use of the `sealed` keyword.